### DISCUSSION TITLE
AssertionError: num_effective_subcarriers must be an integer multiple of num_tx*num_streams_per_tx

### DISCUSSION THREAD

**Discussion by Felix-willaumez** [NVlabs/sionna#677] [Q&A] [answered]
Hello Sionna team,

I am trying to implement a MIMO OFDM scenario in which there is 1 base station with multiple antennas that is communicating (downlink) with different users. 
Doing so I faced an error telling me that the number of subcarriers should be equal to the number of tx (1 base station in my case) * num of streams transmitted by the tx (number of users). I was able to make the code work by playing with the fft size and the number of null tones but I was wondering where does this requirement comes from. Indeed I fail to understand what could be the justification from a communication system point of view. 

Thank you for your attention.

**Comment by jhoydis** [comment]
Hi @Felix-willaumez,

We would need a small code snippet that allows reproducing the behavior.

**Reply by Felix-willaumez** [reply]
Thank you for your quick answer @jhoydis . Here are some more details on the mentioned error.
I want to have 1 base station that is doing beamforming to 8 users and each stream/beam  is made of  `num_effective_subcarriers` (= 1024 - 25 - 26 -1 =972) so I created the following ressource grid.

```py
self._rg = ResourceGrid(num_ofdm_symbols= 14,
                                fft_size= 1024,
                                subcarrier_spacing = 15e3,
                                num_tx = 1, 
                                num_streams_per_tx = 8,
                                cyclic_prefix_length= 16,
                                num_guard_carriers= [25,26],
                                dc_null= True,
                                pilot_pattern= "kronecker",
                                pilot_ofdm_symbol_indices= [1,7,12]) 
```

However doing so raises this error: `num_effective_subcarriers` must be an integer multiple of
      `num_tx`*`num_streams_per_tx`. Indeed I understand that 972 is not a multiple of 8 but I do not understand why an OFDM system would require such a restriction. Is it just for coding purposes or is there something that I am missing ?

**Comment by jhoydis** [comment, marked as answer]
Hi @Felix-willaumez,

The requirement comes from KroneckerPilotPattern not from OFDM. You can use any other pilot pattern if you want to work with channel estimation. If this is not needed, you can just turn it of.
